buildroot-v86/README.md

```diff
@@ -0,0 +1,199 @@
+# Buildroot V86
+
+We use a custom embedded Linux distribution meant to be run under emulation
+in the browser via [v86](https://github.com/copy/v86).  We build this via
+[Buildroot](https://buildroot.org/) in order to produce:
+
+* a custom Linux 4.15 kernel, which strips out many unnecessary drivers, modules, etc. and adds Plan 9 filesystem sharing
+* a root filesystem and Unix commands via [BusyBox](https://busybox.net/)
+* an ISO-based bootloader (i.e., we create a "DVD" that is booted by v86)
+
+Following the [Buildroot customization docs](https://buildroot.org/downloads/manual/manual.html#customize)
+we create a folder `buildroot-v86/` with all the necessary config files,
+filesystem overlay, and scripts necessary to build our distribution.
+
+## Running via Docker
+
+To build the Docker image use `npm run v86:setup` or manually with:
+
+```bash
+$ docker build -t buildroot .
+```
+
+And then to run the build use `npm run v86:build` or manually with:
+
+```bash
+$ docker run \
+    --rm \
+    --name build-v86 \
+    -v $PWD/v86-out:/build \
+    -v $PWD/buildroot-v86/:/buildroot-v86 \
+    buildroot
+```
+
+NOTE: we define two [volumes](https://docs.docker.com/engine/reference/builder/#volume) to 
+allow the container to access our v86 config, and also to write the ISO once complete.  In the
+above I've used `$PWD`, but you can use any absolute path.
+
+When the build completes, a build ISO file will be places in `./v86-out/v86-linux.iso`
+in your source tree (i.e., outside the container).
+
+If you need to re-configure things instead of just running the build, do the following:
+
+```bash
+$ docker run \
+    --rm \
+    --name build-v86 \
+    -v $PWD/v86-out:/build \
```

Comment:
Instead of mounting through a volume, we can just fetch necessary files through [COPY](https://docs.docker.com/engine/reference/builder/#copy). In my opinion, the only time we would like to use volumes, is when we want to share the same content between multiple containers.

Reply:
The reason I don't want to `COPY` them into the container at build time is that I will likely want to edit them (e.g., play with the configs), so mounting them in let's me modify things easily in future. so we don't have to rebuild unnecessarily.

Reply:
In this case you are forcing all containers to share the same config files. Sometimes, that is not an intended behavoiur, as you might want to preserve state of a container.
If you want to edit the files, they could be easily [copied](https://stackoverflow.com/questions/22907231/copying-files-from-host-to-docker-container) over to the container in runtime without the need to rebuild it.
I guess, this is a question of preference, as both of the approaches could be played around to achieve what you want.